### DISCUSSION TITLE
Next major (5.x.x)

### DISCUSSION THREAD

**Discussion by firecow** [firecow/gitlab-ci-local#335] [Ideas]
- Use yargs cmdDir functionality
  - `gitlab-ci-local pipeline`
  - `gitlab-ci-local run <...jobname>`
  - `gitlab-ci-local help`
  - `gitlab-ci-local list`
  - `gitlab-ci-local preview`
  - etc. etc. etc.

- Make `--needs` default
  - It reduces debugging significantly, but increases project bootstrapping
  - `--no-needs` can be used to opt-out

- Make `--shell-isolation` default
  - Bugs are found earlier in `shell-executor` jobs.
  - `--no-shell-isolation` can be used to opt out if projects are actively utilizing it.
  
- Remove `.gitlab-ci-local.yml` special handling
  - Developers can use `rules: [ { if: $GITLAB_CI == 'false', when: on_success } ]` or `GCL_FILE=.gitlab-ci-custom.yml`

- No more auto guessing file variables `~/.gitlab-ci-local/variables.yml`. Explicitly set a `type: file` flag with the key/value pair

- `@NoArtifactsToHost` should be inversed

- Use `.env` extension instead of appending `-env`

**Comment by firecow** [comment]
I'm planning to make a few breaking changes, please stop me or give me a go ahead (:+1:)

@lvjp @solidgoldbomb @JoyceBabu @bcouetil @aepfli @moberghammer @nizacegodk @ColdIce1 @jboyens @jeppenodgaard @atli-c @pcuci

**Comment by bcouetil** [comment]
Agreed, that will simplify my workflow 🤞

**Comment by pcuci** [comment]
I have the following setup in all my projects make the NEEDS flag more accessible, so the proposal makes sense to me.

![image](https://user-images.githubusercontent.com/2169849/138996968-6cfee298-ac43-456d-b97e-7e83ee8e76b6.png)

**Reply by firecow** [reply]
@pcuci Btw, you can receive the same result now, by adding envvar exports to your `.bashrc`
```
export GCL_NEEDS='true' >> ~/.bashrc
export GCL_FILE='.gitlab-ci-local.yml' >> ~/.bashrc
```

**Comment by hverlin** [comment]
Should we suggest to set `--no-artifacts-to-source` by default in the readme as well to prepare for the next major release?

```
export GCL_NEEDS="true"
export GCL_SHELL-ISOLATION="true"
export GCL_ARTIFACTS_TO_SOURCE="false"
```

**Reply by firecow** [reply]
Nah, not yet...

As of right now, there are no real reason to even publish a new major release.
